### ISSUE TITLE
Reset / Clear buttons should have a consistent styling

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#23888]
**Describe the bug**
Various user interface components display a "Reset" or "Clear" button. These buttons have at least 3 different styles. For better accessibility and UI consistency, they all should use the same styling.

**To reproduce**
1 WordPress 5.5
2. observe for example the following button:
  - paragraph block -> sidebar -> typography -> font size Reset button
  - paragraph block -> sidebar -> color settings -> custom color Clear button
  - image block -> sidebar -> image dimensions -> Reset button

**Expected behavior**
- these buttons to have the same styling
- also, for accessibility, they need to be identified as user interface controls so they need a border: the image dimensions -> Reset button looks like text

**Screenshots**

<img width="800" alt="Screenshot 2020-07-12 at 13 26 16" src="https://user-images.githubusercontent.com/1682452/87246296-d5b21d80-c44c-11ea-825a-1876e1097c70.png">

**Editor version (please complete the following information):**
- WordPress version: 5.5 trunk
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? default

**Comment by afercia** [comment]
Additionally: the buttons font-size of `11px` is really to small. This applies to all the `is-small` buttons and should be improved holistically.

**Comment by folletto** [comment]
Thanks for finding this out.
It's definitely something we should look into, especially the alone "Reset" that doesn't feel even clickable.

**Comment by noahshrader** [comment]
Agreed. Thanks for catching these!

For the button referenced in the first column, those are both the "secondary" button variation, just in different states (the first is disabled; the second is resting). The difference is that the first one is given a custom height to match the vertical height of the inputs, which makes sense. Setting that to the default height of a small button (which is 6px shorter) looks imbalanced.

I played around with bringing that 30px height to the "Clear" buttons, but that's going to start extending the exception out which will lead to more inconsistencies. Maybe we do take another look at the small button styles.

With the "Reset" button, I quickly put together a couple of screenshots: one with the secondary button styles applied (which may or may not blend with the options to its left), and one with the tertiary button style. Also, brought it closer in proximity to the options it's resetting.

<img width="337" alt="Screen Shot 2020-07-16 at 1 50 20 PM" src="https://user-images.githubusercontent.com/3665539/87711820-6d757b80-c76d-11ea-938d-3fa2642221e7.png">

<img width="328" alt="Screen Shot 2020-07-16 at 2 02 19 PM" src="https://user-images.githubusercontent.com/3665539/87712857-eaedbb80-c76e-11ea-9899-330cb009247e.png">

>Additionally: the buttons font-size of 11px is really too small
Did you have any recommendations here? 12px?

**Comment by afercia** [comment]
> For the button referenced in the first column, those are both the "secondary" button variation, just in different states (the first is disabled; the second is resting).

While disabled controls don't have contrast requirements, I'm not sure entirely removing the _shape_ of a button when it's disabled makes things any better. Without the shape of a button, this control can't be even identified as a button. It looks like just text. I do realize this is part of a broader issue though, and should be discussed separately. See also #23890.

**Comment by youknowriad** [comment]
All block controls are now using tool panels, so this is mostly solved. 

Now, there are still some controls that are outside the inspector (in drop-downs) but I think for these the UI is different enough that it doesn't need to match ToolPanels.
